WordPress/gutenberg
# [Menu items REST API] Perform validation and sanitization separately from saving

This PR removes the `prepare_item_for_database` method in favor of more specific validation and sanitization callbacks, with the ultimate goal being supporting batch requests. Let's discuss in comments!

Test plan:
1. Apply https://github.com/WordPress/wordpress-develop/pull/311 to your local WordPress.
1. Manually add support for `sanitize_callback` (see comments on the PR above).
1. Run phpunit tests, confirm they pass.
